difflint/lint.py

```diff
@@ -0,0 +1,62 @@
+import os.path
+from pkg_resources import resource_filename
+import shutil
+import subprocess
+
+
+def check_linters():
+    """Checks that all required external executables are in the PATH. Returns a
+    list of executables that are not, or an empty list if everything is OK."""
+    required = ['jscs', 'jshint']
+    return [cmd for cmd in required if shutil.which(cmd) is None]
+
+
+class LintOutput(object):
+    def __init__(self):
+        self.output = ''
+        self._warnings_present = False
+
+    def has_warnings(self):
+        return self._warnings_present
+
+    def get_split_output(self):
+        return self.output.splitlines(keepends=True)
+
+    def run_command(self, args):
+        try:
+            output_bytes = subprocess.check_output(args)
+        except subprocess.CalledProcessError as e:
+            output_bytes = e.output
+            self._warnings_present = True
+        self.output += output_bytes.decode()
+
+
+def _lint_javascript(file_to_lint, mode=None):
+    output = LintOutput()
+    output.run_command(['jscs', file_to_lint, '--reporter',
+        resource_filename(__name__, 'data/jscs_terse_reporter.js')])
```

Comment:
nit: I feel like Pylint recommends vertical alignment in cases like this (multiple instances).